lib/rules-generator.js

```diff
@@ -403,7 +416,15 @@ util.methods(Generator, {
       return exp;
 
     case 'var':
+      var valueType = exp.valueType;
       exp = lookupVar(exp);
+      if (valueType) {
```

Comment:
here I am not sure what the means again. the absent of a value type means what exactly? An <any>? Maybe consider introducing a type label meaning the absece of type, e.g. any